src/coreComponents/physicsSolvers/fluidFlow/wells/SinglePhaseWell.cpp

```diff
@@ -106,14 +108,20 @@ void SinglePhaseWell::validateWellConstraints( MeshLevel const & meshLevel ) con
   {
     WellControls const & wellControls = getWellControls( subRegion );
     WellControls::Control const currentControl = wellControls.getControl();
-    real64 const targetTotalRate = wellControls.getTargetTotalRate();
-    real64 const targetPhaseRate = wellControls.getTargetPhaseRate();
-    GEOSX_ERROR_IF( currentControl == WellControls::Control::PHASEVOLRATE,
-                    "Phase rate control is not available for SinglePhaseWell" );
-    GEOSX_ERROR_IF( isZero( targetTotalRate ),
-                    "Target total rate cannot be equal to zero" );
-    GEOSX_ERROR_IF( !isZero( targetPhaseRate ),
-                    "Target phase rate cannot be used for SinglePhaseWell" );
+    real64 const targetTotalRate = wellControls.getTargetTotalRate( m_currentTime ); // validation done at t = 0
+    real64 const targetPhaseRate = wellControls.getTargetPhaseRate( m_currentTime ); // validation done at t = 0
```

Comment:
I would maybe suggest using an explicit `0.0` instead of `m_currentTime` here (if the intent is to _always_ only validate initial conditions), or passing `time` as a parameter to this function.

Reply:
Done, I now use `0.0` here instead of `m_currentTime`. And I moved the well control validation to the beginning of every time step.